user/user_app.go

```diff
@@ -0,0 +1,68 @@
+package user
+
+type UserAppID string
+type AccountID string
+type AllowlistType string
+
+const (
+	AllowlistTypeContracts  AllowlistType = "contracts"
+	AllowlistTypeMethods    AllowlistType = "methods"
+	AllowlistTypeOrigins    AllowlistType = "origins"
+	AllowlistTypeServices   AllowlistType = "services"
+	AllowlistTypeUserAgents AllowlistType = "user_agents"
+)
+
+type UserApp struct {
+	ID                  UserAppID
+	AccountID           AccountID
```

Comment:
I'm getting kind of confused by all the terminology. Thinking of simplifying it to:

```
type UserApp struct {
	UserId UserId
	AppId AppId
```

Also, is `UserApp` the right term here or is it legacy?

For example, is `GatewayApp` more appropriate where we'd have many users attached to it?

Reply:
ID is not user ID though, it's the identifier for the user app itself. And Account ID is not the app ID.

See attached diagram for the relationship of entities in our current Gateway. [Will attach shortly]